proto/accessgraph/v1alpha/azure.proto

```diff
@@ -0,0 +1,98 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+syntax = "proto3";
+package accessgraph.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/gravitational/teleport/gen/proto/go/accessgraph/v1alpha;accessgraphv1alpha";
+
+// AzureResourceList is a list of Azure resources
+message AzureResourceList {
+  repeated AzureResource resources = 1;
+}
+
+// AWSResource is a list of AWS resources supported by the access graph.
+message AzureResource {
+  oneof resource {
+    AzurePrincipal principal = 1;
+    AzureRoleDefinition role_definition = 2;
+    AzureRoleAssignment role_assignment = 3;
+    AzureVirtualMachine virtual_machine = 4;
+    AzureManagedDatabase managed_database = 5;
+    AzureAKSCluster aks_cluster = 6;
+  }
+}
+
+// AzureVirtualMachine is an Azure virtual machine
+message AzureVirtualMachine {
+  string id = 1;
+  string subscription_id = 2;
+  google.protobuf.Timestamp last_sync_time = 3;
+  string name = 4;
```

Comment:
Include VM tags 
Include the resource group as well as it will be mandatory for the graph display